File: 3-Business/2-Core/lm.Comol.Core.DomainModel/Domain/Repository/Old/BaseFile.vb
Namespace lm.Comol.Core.DomainModel
	<Serializable(), CLSCompliant(True)> Public Class BaseFile
		Inherits lm.Comol.Core.DomainModel.DomainObject(Of System.Guid)
		Implements iBaseFile

#Region "Private"
		Private _Name As String
		Private _Extension As String
		Private _Description As String
		'Private _FileSystemName As String
		Private _Size As Long
		Private _ContentType As String
		Private _MetaData As MetaData
		Private _Owner As Person
		Private _HardLink As Long
		Private _IsDownloadable As Boolean
		Private _IsScormFile As Boolean
#End Region

#Region "Public Overridable Property"
        Public Overridable Property Name() As String Implements iBaseFile.Name
            Get
                Return Me._Name
            End Get
            Set(ByVal value As String)
                Me._Name = value
            End Set
        End Property
        Public Overridable ReadOnly Property DisplayName() As String Implements iBaseFile.DisplayName
            Get
                Return Me._Name & Me._Extension
            End Get
        End Property
        Public Overridable Property Extension() As String Implements iBaseFile.Extension
            Get
                Return Me._Extension
            End Get
            Set(ByVal value As String)
                Me._Extension = value
            End Set
        End Property
        Public Overridable Property Description() As String Implements iBaseFile.Description
            Get
                Return Me._Description
            End Get
            Set(ByVal value As String)
                Me._Description = value
            End Set
        End Property
        Public Overridable ReadOnly Property FileSystemName() As String Implements iBaseFile.FileSystemName
            Get
                Return MyBase.Id.ToString & ".stored"
            End Get
        End Property
        Public Overridable Property Size() As Long Implements iBaseFile.Size
            Get
                Return Me._Size
            End Get
            Set(ByVal value As Long)
                Me._Size = value
            End Set
        End Property
        Public Overridable Property ContentType() As String Implements iBaseFile.ContentType
            Get
                Return Me._ContentType
            End Get
            Set(ByVal value As String)
                Me._ContentType = value
            End Set
        End Property
        Public Overridable Property MetaInfo() As iMetaData Implements iBaseFile.MetaInfo
            Get
                If IsNothing(Me._MetaData) Then
                    Me._MetaData = New lm.Comol.Core.DomainModel.MetaData
                End If
                Return Me._MetaData
            End Get
            Set(ByVal value As iMetaData)
                Me._MetaData = value
            End Set
        End Property
        Public Overridable Property Owner() As iPerson Implements iBaseFile.Owner
            Get
                Return Me._Owner
            End Get
            Set(ByVal value As iPerson)
                Me._Owner = value
            End Set
        End Property
        Public Overridable Property HardLink() As Long Implements iBaseFile.HardLink
            Get
                Return Me._HardLink
            End Get
            Set(ByVal value As Long)
                Me._HardLink = value
            End Set
        End Property
        Public Overridable ReadOnly Property SizeKB() As Double Implements iBaseFile.SizeKB
            Get
                Return Math.Round(_Size / 1024, 2)
            End Get
        End Property

        Public Overridable ReadOnly Property SizeMB() As Double Implements iBaseFile.SizeMB
            Get
                Return Math.Round(_Size / (1024 * 1024), 2)
            End Get
        End Property
        Public Overridable Property IsDownloadable() As Boolean Implements iBaseFile.IsDownloadable
            Get
                Return _IsDownloadable
            End Get
            Set(ByVal value As Boolean)
                _IsDownloadable = value
            End Set
        End Property
        Public Overridable Property IsScormFile() As Boolean Implements iBaseFile.IsScormFile
            Get
                Return _IsScormFile
            End Get
            Set(ByVal value As Boolean)
                _IsScormFile = value
            End Set
        End Property
#End Region

		Sub New()
			Me._IsScormFile = False
			Me._IsDownloadable = False
		End Sub
	End Class
End Namespace

File: 3-Business/2-Core/lm.Comol.Core.DomainModel/Domain/Repository/Old/BaseFile.cs

using System;
namespace lm.Comol.Core.DomainModel
{
	[Serializable(), CLSCompliant(true)]
	public class BaseFile : DomainObject<System.Guid>, iBaseFile
	{
        private iMetaData _MetaData;
		public virtual string Name{ get; set; }
		public virtual string DisplayName {
			get { return this.Name + this.Extension; }
		}
		public virtual string Extension { get; set; }
		public virtual string Description { get; set; }
		public virtual string FileSystemName {
			get { return base.Id.ToString() + ".stored"; }
		}
		public virtual long Size{ get; set; }
		public virtual string ContentType{ get; set; }
		public virtual iMetaData MetaInfo {
			get {
				if ((this._MetaData == null)) {
					this._MetaData = new MetaData();
				}
				return this._MetaData;
			}
			set { this._MetaData = value; }
		}
		public virtual iPerson Owner{ get; set; }
		public virtual long HardLink { get; set; }
		public virtual double SizeKB {
			get { return Math.Round((double)Size / 1024, 2); }
		}

		public virtual double SizeMB {
            get { return Math.Round((double)Size / (1024 * 1024), 2); }
		}
		public virtual bool IsDownloadable{ get; set; }
        public virtual bool IsScormFile { get; set; }

		public BaseFile()
		{
			this.IsScormFile = false;
			this.IsDownloadable = false;
		}
	}
}